lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -698,14 +697,22 @@ llvm::json::Value CreateSource(llvm::StringRef source_path) {
   return llvm::json::Value(std::move(source));
 }
 
-static std::optional<llvm::json::Value> CreateSource(lldb::SBFrame &frame) {
+static llvm::json::Value CreateSource(lldb::SBFrame &frame,
```

Comment:
I assume this function is used elsewhere? The line number and column have a different meaning based on whether we have source info and having the logic spread across this function and `CreateStackFrame` seems like a pretty big foot gun. Any chance we can localize this logic?

Reply:
Merged this into the only place that called the helper (`CreateStackFrame`) to make sure its more clear.